adaptive/learner/balancing_learner.py

```diff
@@ -173,6 +178,20 @@ def _ask_and_tell_based_on_npoints(self, n):
         points, loss_improvements = map(list, zip(*selected))
         return points, loss_improvements
 
+    def _ask_and_tell_based_on_cycle(self, n):
+        if not hasattr(self, "_cycle"):
+            self._cycle = itertools.cycle(range(len(self.learners)))
```

Comment:
do this in `__init__` as I commented above.

Reply:
But you wrote to do in the strategy property setter?